lib/apiservers/engine/backends/container.go

```diff
@@ -415,6 +417,41 @@ func (c *Container) ContainerRm(name string, config *types.ContainerRmConfig) er
 	return nil
 }
 
+// checkActivePortBindings gets port bindings for the container and
+// unmaps ports if the cVM that previously bound them isn't powered on
+func (c *Container) checkActivePortBindings(vc *viccontainer.VicContainer, endpoints []*models.EndpointConfig) error {
+	for p, _ := range vc.HostConfig.PortBindings {
+		thePort := p.Port()
+		if _, mapped := containerByPort[thePort]; mapped {
+			mappedCtr := containerByPort[thePort]
+			log.Debugf("Container %q maps port %s", mappedCtr, thePort)
+			// check state of the previously bound container with PL
+			cc := cache.ContainerCache().GetContainer(mappedCtr)
+			running, err := c.containerProxy.IsRunning(cc)
+			if err != nil {
+				return fmt.Errorf("Failed to get container %q power state: %s", mappedCtr, err)
+			}
+			if !running {
+				// unmap ports if not powered on
+				log.Debugf("Unmapping port %s for powered off container %q", thePort, mappedCtr)
+				if len(endpoints) < 1 {
+					return fmt.Errorf("Failed to unmap port %s for container %q: no endpoints", thePort, mappedCtr)
```

Comment:
Is this necessary to unmap? If we are keeping track of the iptables commands, we don't really need endpoint info about the container.